### PR TITLE
Add tests to improve coverage

### LINKED ISSUE #228: Add more Testcases to increase Code Coverage

### PR DESCRIPTION
**Pull request by aidoxe-123**

### CODE REVIEW

File: src/main/java/jimmy/mcgymmy/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -54,9 +54,6 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
     public static Name parseName(String name) throws ParseException {
         requireNonNull(name);
         String trimmedName = name.trim();
-        if (!Name.isValidName(trimmedName)) {
-            throw new ParseException(Name.MESSAGE_CONSTRAINTS);
-        }
```
Comment (Jh123x) [inline src/main/java/jimmy/mcgymmy/logic/parser/ParserUtil.java:59]: Why was this removed?

File: src/test/java/jimmy/mcgymmy/model/UserPrefsTest.java [hunk 2]
```diff
@@ -19,4 +21,13 @@ public void setMcGymmyFilePath_nullPath_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> userPrefs.setMcGymmyFilePath(null));
     }
 
+    @Test
+    public void equals_returnsCorrectResult() {
+        UserPrefs userPref = new UserPrefs();
+        // identical -> returns true
+        assertEquals(userPref, userPref);
```
Comment (Jh123x) [inline src/test/java/jimmy/mcgymmy/model/UserPrefsTest.java:28]: I think we should add a test case where there are 2 userPrefs with the same values but not the same object.

**Review by Jh123x — CHANGES_REQUESTED** [review]
Mostly LGTM ><

Overlaps with #230

File: src/main/java/jimmy/mcgymmy/logic/parser/ParserUtil.java (hunk 1 above)
Comment (aidoxe-123) [reply src/main/java/jimmy/mcgymmy/logic/parser/ParserUtil.java:59]: that code and this one have the same functionality:
image
So I just remove 1 of them

Comment (Jh123x) [reply src/main/java/jimmy/mcgymmy/logic/parser/ParserUtil.java:59]: Might be better to leave it in for defensive programming?
Edit: Sure that works too

File: src/test/java/jimmy/mcgymmy/model/UserPrefsTest.java (hunk 2 above)
Comment (aidoxe-123) [reply src/test/java/jimmy/mcgymmy/model/UserPrefsTest.java:28]: added!

**Review by Jh123x — APPROVED** [review]
LGTM
